### DISCUSSION TITLE
Can't block a domain name !

### DISCUSSION THREAD

**Discussion by gcpmusic** [DNSCrypt/dnscrypt-proxy#1712] [Q&A] [answered]
I can't manage to completely block sentry.io domain even with .sentry. as show below. Any suggestion?
Thank you in advance.

<img width="419" alt="Screenshot 2021-05-11 at 21 50 41" src="https://user-images.githubusercontent.com/32528176/117878087-a7d47a00-b2a5-11eb-81c7-cfebc86c5377.png">

**Comment by lifenjoiner** [comment]
By my test on Windows:
1. Uncomment this line (your config file)
   https://github.com/DNSCrypt/dnscrypt-proxy/blob/31f4d7aa03e54b4c6778312c340e51297da17ded/dnscrypt-proxy/example-dnscrypt-proxy.toml#L510
   to
   ```
     blocked_names_file = 'blocked-names.txt'
   ```
2. Remane `https://github.com/DNSCrypt/dnscrypt-proxy/blob/master/dnscrypt-proxy/example-blocked-names.txt` to `blocked-names.txt`
3. Append `sentry.io` to `blocked-names.txt`

It is blocked!
![blocked_names](https://user-images.githubusercontent.com/19585474/117902728-b2702d00-b300-11eb-9a6d-c7d57315b776.png)

**Reply by gcpmusic** [reply]
Many thanks for the reply. What you suggest, is what I am doing successfully from years and all domain have always been blocked as expected. However, sentry.io is a little devil, as my screenshot above show, in some way sometimes manage to slip away. 

What I don't understand is why the domain is marked with different entry such  HTTPS  and  A . Probably because are a different kind of connections?

In my blocked-names.txt filter, is appended as this:
x.sentry.x    (x stands for asterisk)
sentry.io	HTTPS
sentry.io	A

and this is my new log!!

[2021-05-11 13:25:57]	127.0.0.54	sentry.io	HTTPS	
PASS	172ms	cs-rome
[2021-05-11 13:25:57]	127.0.0.54	sentry.io	A     	
PASS	202ms	cs-rome
[2021-05-11 13:44:36]	127.0.0.54	o36975.ingest.sentry.io	A	
REJECT	1ms	-

Also to note, this connection is called from some of the application in my computer as I never browse the website.

**Reply by lifenjoiner** [reply]
Oops! If it is a special case, you'd better provide your configs first as more as possible, to let others reproduce and/or diagnose what the problem really is.

<br>

> In my blocked-names.txt filter, is appended as this:
> x.sentry.x (x stands for asterix)
> sentry.io HTTPS
> sentry.io A

I think maybe it's because of the syntaxes are only:
https://github.com/DNSCrypt/dnscrypt-proxy/blob/31f4d7aa03e54b4c6778312c340e51297da17ded/dnscrypt-proxy/example-blocked-names.txt#L6-L16

**Comment by gcpmusic** [comment]
Thanks for the reply. The syntax is correct, everything is working perfectly here a part for sentry.io block. 
As here the asterisk seem not allowed, I replaced with x as explained in the brackets but just created confusion. 

A screenshot of my blocked-names.txt is better:

<img width="156" alt="Screenshot 2021-05-12 at 14 30 11" src="https://user-images.githubusercontent.com/32528176/117976390-05150d80-b330-11eb-9128-f0cb4cd1ace4.png">

This is my full .toml config as request.
[dnscrypt-proxy.toml.zip](https://github.com/DNSCrypt/dnscrypt-proxy/files/6466252/dnscrypt-proxy.toml.zip)

Thank you for the help. Really appreciated.

**Comment by lifenjoiner** [comment]
After reading 
https://github.com/DNSCrypt/dnscrypt-proxy/blob/eda8dd51817cb82fc32d42e18c50d667c4b9f6a8/dnscrypt-proxy/plugin_block_name.go#L95-L122
and
https://github.com/DNSCrypt/dnscrypt-proxy/blob/eda8dd51817cb82fc32d42e18c50d667c4b9f6a8/dnscrypt-proxy/pattern_matcher.go

It seems only `@weeklyRanges` is possible as the 2nd part of an entry; neither `HTTPS`, nor `A`.
You can also see the examples at the end of `example-blocked-names.txt`.

**Comment by gcpmusic** [comment]
Yes I know. I only added later 

sentry.io      HTTPS  
and
sentry.io      A   

in the blocked-names.txt as showed in the log just because the correct pattern  asterisk.sentry.asterisk doesn't work. Note that only for this domain this pattern doesn't work. For other hundreds domain this and all other patterns works perfectly.
Thanks

**Comment by jedisct1** [comment]
`*.sentry.*` should work like any other pattern.

Check that all characters are correct, thank you don't have a `,` instead of a dot, or a unicode character like `⁎` instead of `*` . Maybe delete the line and type it again.

Also check that it's not overridden by a rule in the allow list.

Or share the actual block file, since screenshots are not very helpful to reproduce this.

**Comment by ghost** [comment, marked as answer]
Hi!
Am I missing something? From the code it seems like the asterisks are trimmed and then it is checked if "sentry.io" contains ".sentry." which it doesn't. So everything seems to work as expected.

**Reply by jedisct1** [reply]
Ah, of course. I didn't look at the code, but obviously if you want to match `yz`, blocking `*xyz*` is not going to do anything.
Same thing for `sentry` vs `*.sentry*`. If you're matching substrings, a dot character is just the same thing as any other character.

**Comment by gcpmusic** [comment]
livingentity,
Yes the asterisks are trimmed here but are properly in place in the blocked-names.txt and everything isnt fine, works partially.

jedisct1,
I already did all the check you suggested. Thanks
Here is the actual block file log as requested:
[query.log.zip](https://github.com/DNSCrypt/dnscrypt-proxy/files/6468716/query.log.zip)

Anyway I just realised that this happen only on Big Sur. On Mojave is fine but I think only because there is not app in my Mojave that trigger the sentry.io DNS call.

I suspect is the LuLu firewall requesting the connection but the issue still remains as should block it anyway fully and not partially as the log show.

**Reply by jedisct1** [reply]
If you want to match the `.sentry.` substring, write `*.sentry.*`. `sentry.io` doesn't contain the `.sentry.` substring.

If you want to match anything that starts with `sentry.` such as `sentry.io`, write `sentry.*`.

If you want to match `sentry.io`, and not `www.sentry.io`, write `=sentry.io`.

**Reply by gcpmusic** [reply]
Thanks, I am well aware of all the pattern combination. I use DNSCript since v1.
I am using `*.sentry.*` which should block any query related as do correctly from plenty other domain I block. On this one for some reason the block is partial as per the attached log above.

**Reply by jedisct1** [reply]
`sentry.io` doesn't contain `.sentry.`.
`sentry.io HTTP` is not valid syntax.

So, `sentry.io` cannot be blocked.

**Reply by gcpmusic** [reply]
jedisct1 thank you. I am not sure to understand. 

The - HTTPS -  and the - A - next to sentry.io show up only for some of the sentry.io query on the log file and these are the only non rejected query.

All the other query are blocked such:
www.sentry.io
www.sentry.dev
o36975.ingest.sentry.io

I don't know what the added - HTTPS -  and the - A - mean. However, I notice that they also appears on other domain that actually I don't block, on some that I do block and on all the nx.log which, for my understanding, don't go anywhere. Am I correct? 

Totally confused 🤔

**Reply by jedisct1** [reply]
> All the other query are blocked such:
> `www.sentry.io`
> `www.sentry.dev`
> `o36975.ingest.sentry.io`

You are blocking `.sentry.`

The names above contain `.sentry.`

`sentry.io`, without any `.` before `sentry` doesn't contain `.sentry.`.

**Reply by gcpmusic** [reply]
I see, many many thanks. `sentry.*` works 👍  and together `*.sentry.*` all sentry.io are finally blocked.
Cheers

**Reply by jedisct1** [reply]
`*🍉*` does the same thing as the "search" function of your text editor when searching for 🍉 

Open a text editor.

Type:

```text
sentry.io
foo.sentry.bar
```

Then use the search function to search for `.sentry.` - The text editor says that there is no match on the first line, only on the second.

**Reply by gcpmusic** [reply]
Great tip thanks. In the future I will use a text editor as debugger :-)
